packages/application-shell/src/components/navbar/navbar.js

```diff
@@ -379,111 +379,131 @@ export class DataMenu extends React.PureComponent {
     return NO_VALUE_FALLBACK;
   };
 
-  render() {
+  renderMenu = (menu, menuType, index) => {
+    const isActive = this.state.activeItemIndex === `${menuType}-${index}`;
+    const MenuIcon = Icons[menu.icon];
+    const hasSubmenu = Boolean(menu.submenu) && menu.submenu.length > 0;
     return (
-      <MenuGroup level={1}>
-        {this.props.data.map((menu, index) => {
-          const isActive = this.state.activeItemIndex === index;
-          const MenuIcon = Icons[menu.icon];
-          const hasSubmenu = Boolean(menu.submenu) && menu.submenu.length > 0;
-          return (
-            <ToggledWithPermissions
-              key={menu.key}
-              featureToggle={menu.featureToggle}
-              permissions={menu.permissions}
+      <ToggledWithPermissions
+        key={menu.key}
+        featureToggle={menu.featureToggle}
+        permissions={menu.permissions}
+      >
+        <React.Fragment>
+          {menu.key === 'Settings' && <MenuItemDivider />}
+          <MenuItem
+            hasSubmenu={hasSubmenu}
+            isActive={isActive}
+            isMenuOpen={this.state.isMenuOpen}
+            onClick={() => {
+              this.handleToggleItem(menuType, index);
+            }}
+            onMouseEnter={
+              this.state.isMenuOpen
+                ? null
+                : () => this.handleToggleItem(menuType, index)
```

Comment:
Nitpick: Can we early-return when `this.state.isMenuOpen` is `true` in `handleToggleItem`, instead of  conditionally adding this prop

Reply:
I would say conditionally adding the prop might be preferable as we don't keep a `mouseEnter` event listener around in case the menu is open already.